allennlp/models/bimpm.py

```diff
@@ -0,0 +1,193 @@
+"""
+BiMPM (Bilateral Multi-Perspective Matching) model implementation.
+"""
+
+from typing import Dict, Optional, List, Any
+
+from overrides import overrides
+import torch
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.data import Vocabulary
+from allennlp.modules import FeedForward, Seq2SeqEncoder, Seq2VecEncoder, TextFieldEmbedder
+from allennlp.models.model import Model
+from allennlp.nn import InitializerApplicator, RegularizerApplicator
+from allennlp.nn import util
+from allennlp.training.metrics import CategoricalAccuracy
+
+from allennlp.modules.bimpm_matching import BiMPMMatching
+
+
+@Model.register("bimpm")
+class BiMPM(Model):
+    """
+    This ``Model`` implements BiMPM model described in `Bilateral Multi-Perspective Matching
+    for Natural Language Sentences <https://arxiv.org/abs/1702.03814>`_ by Zhiguo Wang et al., 2017.
+    Also please refer to the `TensorFlow implementation <https://github.com/zhiguowang/BiMPM/>`_ and
+    `PyTorch implementation <https://github.com/galsang/BIMPM-pytorch>`_.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the ``premise`` and ``hypothesis`` ``TextFields`` we get as input to the
+        model.
+    word_matcher : ``BiMPMMatching``
+        BiMPM matching on the output of word embeddings of premise and hypothesis.
+    encoder1 : ``Seq2SeqEncoder``
+        First encoder layer for the premise and hypothesis
+    matcher_fw1 : ``BiMPMMatching``
```

Comment:
Instead of `fw` and `bw`, can you expand those to `forward` and `backward`?  It's much easier to read.